docs/SPEC.md

```diff
@@ -1249,18 +1309,30 @@ The _null_ value can be of any data type.
 ### Transformations
 
 Transformations define a change to a stream.
-Transformations may consume an input stream and always produce a new output stream.
+Transformations must consume an input stream and always produce a new output stream.
 
-Most transformations output one table for every table they receive from the input stream.
-Transformations that modify the group keys or values will need to regroup the tables in the output stream.
-A transformation produces side effects when it is constructed from a function that produces side effects.
+Transformations always operate on the data channel within the stream.
+Transformations do not operate on the meta channel within the stream, however they may add new tables to it.
```

Comment:
These two sentences seem a bit inconsistent. `meta` and `errors` consume from the meta channel directly. And all transformations will have the ability to add tables to the meta channel in case of errors. Perhaps describe what the meta channel is used for but remove these two sentences.

Reply:
I think that what you want to say is that user-defined behavior in transformations (like high-order functions) cannot arbitrarily manipulate meta-channels. Meta channels are manipulated by the transformations as a side-effect of applying the user-defined behavior, let it be errors produced by UDF application, statistics about performance, or else.